File: deploy/utils/listenEvents.ts
import  { Wallet, utils,Provider } from "zksync-web3";
const ethers = require('ethers');
const provider = new Provider("https://zksync2-testnet.zksync.dev");

// 设置你的合约地址和ABI
let contractAddress = "your_contract_address";
let abi = ["Your ABI"];

// 创建合约实例
let contract = new ethers.Contract(contractAddress, abi, provider);

// 实时监听"ListingForSale"事件
contract.on("ListingForSale", (seller, tokenId, price, event) => {
    console.log(`Block: ${event.blockNumber}, Seller: ${seller}, Token ID: ${tokenId.toString()}, Price: ${price.toString()}`);
    // 在这里，你可以把事件的信息存储到你的数据库
});

File: deploy/utils/cs/ListenEvents.cs
using System.Threading.Tasks;
using Nethereum.Contracts;
using Nethereum.Web3;

public class ListenEvents
{
    // 你的合约地址和ABI
    private static string contractAddress = "your_contract_address";
    private static string abi = "Your ABI";

    public static async Task Main(string[] args)
    {
        // 创建Web3实例
        Web3 web3 = new Web3("https://mainnet.infura.io");

        // 创建合约实例
        Contract contract = web3.Eth.GetContract(abi, contractAddress);

        // 创建过滤器
        var filterAll = await contract.Event("ListingForSale").CreateFilterAsync();

        // 循环查询新的事件
        while (true)
        {
            var log = await contract.Event("ListingForSale").GetFilterChanges<ListingForSaleEventDTO>(filterAll);
            foreach (var e in log)
            {
                // 在这里，你可以把事件的信息存储到你的数据库
                System.Console.WriteLine($"Block: {e.Log.BlockNumber.Value}, Seller: {e.Event.Seller}, Token ID: {e.Event.TokenId}, Price: {e.Event.Price}");
            }

            // 等待一段时间后再查询新的事件
            await Task.Delay(10000);
        }
    }
}
